### ISSUE TITLE
Guzzle problem

### ISSUE THREAD

**Issue by gmatkowski** [matthewbdaly/laravel-azure-storage#23]
Problem 1
    - Installation request for matthewbdaly/laravel-azure-storage ^1.4 -> satisfiable by matthewbdaly/laravel-azure-storage[1.4.0].
    - Can only install one of: guzzlehttp/guzzle[6.5.x-dev, 7.2.0].
    - Can only install one of: guzzlehttp/guzzle[7.2.0, 6.5.x-dev].
    - Can only install one of: guzzlehttp/guzzle[6.5.x-dev, 7.2.0].
    - Conclusion: install guzzlehttp/guzzle 6.5.x-dev
    - Installation request for guzzlehttp/guzzle (locked at 7.2.0, required as ^7.0.1) -> satisfiable by guzzlehttp/guzzle[7.2.0].

So it looks like it's not supported for latest Guzzle 7.2.0 version

**Comment by matthewbdaly** [comment]
Looks like a duplicate of #16.

**Comment by gmatkowski** [comment]
> Looks like a duplicate of #16.

yes, You are right sorry for duplicating ...

**Comment by matthewbdaly** [comment]
I will keep an eye on the upstream package and update this package once a suitable version becomes available.

**Comment by matthewbdaly** [comment]
Going to leave this open to prevent more duplication and indicate it's not been forgotten.

**Comment by brpetrov** [comment]
Hi there, Sorry for my question, but as Laravel 8.0 doesn't specifically require guzzlehttp 7.0, would downgrading to 6.0 fix the issue?

**Comment by matthewbdaly** [comment]
@brpetrov This package doesn't itself depend on Guzzle. It's just a Laravel wrapper for Flysystem's Azure integration, which does depend on Guzzle.

Downgrading Laravel is an option, but not a great one TBH.

**Comment by brpetrov** [comment]
@matthewbdaly Thanks for responding. How would you recommend going about it without downgrading? I don't see what else..

**Comment by matthewbdaly** [comment]
Yeah, I don't think there's anything much I can do about it until [Azure/azure-storage-common](https://github.com/Azure/azure-storage-common-php) upgrades the Guzzle version they use. [There's a ticket open for this](https://github.com/Azure/azure-storage-common-php/issues/2), but until that's sorted there's not much that can be done.

I'd really prefer that package vendors used [HTTPlug](http://httplug.io/) rather than relying explicitly on Guzzle for that reason since that way it's dependent on an interface rather than a specific package, but it is what it is.

**Comment by matthewbdaly** [comment]
Looks like the November release of the Azure storage PHP SDK will include support for this - see https://github.com/Azure/azure-storage-php/issues/253 for details. Once that's done it should be possible to update this package accordingly.

**Comment by MrThePlague** [comment]
> Looks like the November release of the Azure storage PHP SDK will include support for this - see https://github.com/Azure/azure-storage-php/issues/253 for details. Once that's done it should be possible to update this package accordingly.

Looks like they just merged these changes!

**Comment by matthewbdaly** [comment]
Still waiting on a new release before I can do anything about this...

**Comment by WESquad-oldmember** [comment]
Found a temporary solution on this repo: [squigg/azure-queue-laravel](https://github.com/squigg/azure-queue-laravel)


> Laravel 8 has moved to Guzzle 7.x, but the upstream dependency microsoft/azure-storage-queue from this package still uses Guzzle 6. This will cause composer to fail during dependency resolution.
> 
> Tests so far have not identified any impacting breaking changes between Guzzle 6 and 7, so while we wait for the upstream package to be updated, you can work around this issue by adding/updating your root composer.json file to use an inline alias for guzzlehttp/guzzle:
> ```
> "guzzlehttp/guzzle": "7.0.1 as 6.5.5"
> ```
> Or run this command:
> ```
> composer require guzzlehttp/guzzle:"7.0.1 as 6.5.5" 
> ```

**Comment by Oxicode** [comment]
Vuelve intentar, vi que ya hicieron la actualización por acá
https://github.com/Azure/azure-storage-php/commit/5dc75923f11042bca5b3cb6a16c82f3d104002cd

**Comment by matthewbdaly** [comment]
OK, so there's a new version of the upstream Azure storage PHP SDK, but it looks like we're still waiting for it to be pushed to Packagist. I've made a new release that updates the version dependency on `league/flysystem-azure-blob-storage` and I think this will work once it goes live, but will check.

**Comment by matthewbdaly** [comment]
Looks like this is now resolved. The package `microsoft/azure-storage-blob` now shows as having released version 1.5.2 on Packagist, and adding this package to a new Laravel 8 project works OK. The tests pass too, so it *should* be OK.
